packages/core/test/hardhat/data-verification-mechanism/VotingV2.js

```diff
@@ -5358,6 +5358,47 @@ describe("VotingV2", function () {
     for (const ac of [account1, account2, account3, account4, rand])
       await voting.methods.withdrawRewards().send({ from: ac });
   });
+  it("Changing max rolls while requests are unresolved does not impact requests", async function () {
+    // Consider if requests are resolveable but before they can be resolved the maxRolls variable is changed. If this
+    // was to happen then it is possible that requests that could be resolved get deleted before they resolve. To
+    // protect against this the contract processes resolvable price requests before setting max rolls. To test this
+    // we will construct a price request, vote on the request, move 2 rounds forward, then set maxRolls to a value
+    // that would delete them. Without resolving the request, we will call setMaxRolls. the contract should resolve
+    // these requests before setting the new max rolls.
+
+    const identifier = padRight(utf8ToHex("test"), 64);
+    const time = "1000";
+    await supportedIdentifiers.methods.addSupportedIdentifier(identifier).send({ from: accounts[0] });
+    await voting.methods.requestPrice(identifier, time).send({ from: registeredContract });
+    await moveToNextRound(voting, accounts[0]);
+
+    const price = 123;
+    const salt = getRandomSignedInt(); // use the same salt for all votes. bad practice but wont impact anything.
+    let roundId = (await voting.methods.getCurrentRoundId().call()).toString();
+    let baseRequest = { salt, roundId, identifier };
+    let hash1 = computeVoteHash({ ...baseRequest, price: price, account: account1, time: time });
+    await voting.methods.commitVote(identifier, time, hash1).send({ from: account1 });
+
+    // move to the next phase, reveal and change the slashing library and request another price to vote on in the next round.
+    await moveToNextPhase(voting, accounts[0]);
+    await voting.methods.revealVote(identifier, time, price, salt).send({ from: account1 });
+
+    await moveToNextRound(voting, accounts[0]);
+    await moveToNextRound(voting, accounts[0]);
+
+    assert.equal((await voting.methods.getNumberOfPriceRequests().call()).numberPendingPriceRequests, 1);
+    assert.equal((await voting.methods.getNumberOfPriceRequests().call()).numberResolvedPriceRequests, 0);
+    await voting.methods.setMaxRolls(1).send({ from: accounts[0] });
+
+    assert.equal((await voting.methods.getNumberOfPriceRequests().call()).numberPendingPriceRequests, 0);
+    assert.equal((await voting.methods.getNumberOfPriceRequests().call()).numberResolvedPriceRequests, 1);
+
+    // This action should have resolved the request.
+    assert.equal(
+      (await voting.methods.getPrice(identifier, time).call({ from: registeredContract })).toString(),
```

Comment:
`getPrice` invokes `_getPriceOrError` that resolves price if it was resolvable independent of whether request id was moved from pending to resolved. As long as any other action had not invoked deletion of request this test would succeed, and it does even before the fix in this PR is applied.

Instead the test should demonstrate that resolvable request cannot get deleted after changing `maxRolls` and invoking `processResolvablePriceRequests` after that.